gateway/handler_error.go

```diff
@@ -311,6 +314,29 @@ func (e *ErrorHandler) HandleError(w http.ResponseWriter, r *http.Request, errMs
 			log.WithError(err).Error("could not store analytic record")
 		}
 	}
+
```

Comment:
Expecting one line added here: `e.recordAccessLog(r.Context(), log)`, move all this code into own function. The logger is passed along because otherwise it's a global coupling. The handlers don't yet have their own logger (or one passed down).

Logic here slightly differs, ensure consistent implementation between types, as I understand the logic for both should be identical (invoking ctxGetAuthToken based on ctx).